collab-document/src/importer/md_importer.rs

```diff
@@ -0,0 +1,509 @@
+        constructs: Constructs {
+          math_text: true,
+          math_flow: true,
+          autolink: true,
+          ..Constructs::gfm()
+        },
+        ..ParseOptions::gfm()
+      },
+    )?;
+    let mut document_data = DocumentData {
+      page_id: document_id.to_string(),
+      blocks: HashMap::new(),
+      meta: DocumentMeta {
+        children_map: HashMap::new(),
+        text_map: Some(HashMap::new()),
+      },
+    };
+
+    process_node(
+      &mut document_data,
+      &mdast,
+      None,
+      Some(document_id.to_string()),
+      None,
+    );
+
+    Ok(document_data)
+  }
+}
+
+fn process_node(
+  document_data: &mut DocumentData,
+  node: &mdast::Node,
+  parent_id: Option<String>,
+  block_id: Option<String>,
+  list_type: Option<&str>,
+) {
+  if is_inline_node(node) {
+    process_inline(document_data, node, parent_id);
+    return;
+  }
+
+  if let Some((children, list_type)) = get_list_info(node) {
+    process_children(document_data, children, parent_id, Some(list_type));
+    return;
+  }
+
+  let id = block_id.unwrap_or_else(generate_id);
+
+  let block = create_block(&id, node, parent_id.clone(), list_type);
+
+  document_data.blocks.insert(id.clone(), block);
+
+  update_children_map(document_data, parent_id, &id);
+
+  match node {
+    mdast::Node::Root(root) => {
+      process_children(document_data, &root.children, Some(id.clone()), None)
+    },
+    mdast::Node::Paragraph(para) => {
+      process_children(document_data, &para.children, Some(id.clone()), None)
+    },
+    mdast::Node::Heading(heading) => {
+      process_children(document_data, &heading.children, Some(id.clone()), None)
+    },
+    mdast::Node::BlockQuote(_) | mdast::Node::ListItem(_) => {
+      if let Some(mdast::Node::Paragraph(para)) = node_children(node).and_then(|c| c.first()) {
+        process_children(document_data, &para.children, Some(id.clone()), None);
+      }
+    },
+    mdast::Node::Code(code) => {
+      let mut delta = Delta::new();
+      delta.insert(code.value.clone(), Vec::new());
+      insert_delta_to_text_map(document_data, &id, delta);
+    },
+    mdast::Node::Table(table) => process_table(document_data, table, &id),
+    _ => {},
+  }
+}
+
+fn node_type_to_string(node: &mdast::Node, list_type: Option<&str>) -> String {
+  match node {
+    mdast::Node::Root(_) => "page",
+    mdast::Node::Paragraph(_) => "paragraph",
+    mdast::Node::Heading(_) => "heading",
+    mdast::Node::BlockQuote(_) => "quote",
+    mdast::Node::ListItem(list) => {
+      if list.checked.is_some() {
+        "todo_list"
+      } else {
+        match list_type {
+          Some("numbered_list") => "numbered_list",
+          Some("bulleted_list") => "bulleted_list",
+          _ => "bulleted_list",
+        }
+      }
+    },
+    mdast::Node::Definition(defi) => {
+      let url = defi.url.to_string();
+      if is_image_url(&url) {
+        "image"
+      } else {
+        "link_preview"
+      }
+    },
```

Comment:
Does the Definition include the ImageReference and LinkReference?

https://github.com/syntax-tree/mdast?tab=readme-ov-file#linkreference

https://github.com/syntax-tree/mdast?tab=readme-ov-file#imagereference